lib/src/main/java/uk/gov/di/ipv/cri/kbv/api/util/SOAPMessageWriterHandler.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.di.ipv.cri.kbv.api.util;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.xml.soap.SOAPException;
+import javax.xml.soap.SOAPMessage;
+import javax.xml.ws.handler.MessageContext;
+import javax.xml.ws.handler.soap.SOAPHandler;
+import javax.xml.ws.handler.soap.SOAPMessageContext;
+
+import java.io.IOException;
+import java.util.Set;
+
+import static java.lang.System.out;
+
+public class SOAPMessageWriterHandler implements SOAPHandler<SOAPMessageContext> {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    public boolean handleMessage(SOAPMessageContext smc) {
+        try {
+            return logToSystemOut(smc);
+        } catch (SOAPException e) {
+            throw new RuntimeException(e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public Set getHeaders() {
+        return null;
+    }
+
+    public boolean handleFault(SOAPMessageContext smc) {
+        try {
+            return logToSystemOut(smc);
+        } catch (SOAPException e) {
+            throw new RuntimeException(e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void close(MessageContext context) {}
+
+    /*
+     * Check the MESSAGE_OUTBOUND_PROPERTY in the context
+     * to see if this is an outgoing or incoming message.
+     * Write a brief message to the print stream and
+     * output the message. The writeTo() method can throw
+     * SOAPException or IOException
+     */
+    private Boolean logToSystemOut(SOAPMessageContext smc) throws SOAPException, IOException {
+        Boolean outboundProperty = (Boolean) smc.get(MessageContext.MESSAGE_OUTBOUND_PROPERTY);
+        SOAPMessage message = smc.getMessage();
+
+        if (outboundProperty.booleanValue() && LOGGER.isDebugEnabled()) {
```

Comment:
You'll need to either update the log4j.xml to include debug or remove the `&& LOGGER.isDebugEnabled()` otherwise we might not get any of these outputs.

Reply:
I was thinking of setting this on the questionHandler in AWS

Reply:
If we're okay with clickops / have the permissions to do that.

Reply:
I think it is better this way localizing DEBUG to the questionHandler